ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -452,9 +452,11 @@ const Page: NextPage<Props> = ({
       setSource((src) => {
         src.options[optKey] = optValue;
         src.mapping = {};
+
+        loadPreview(src);
```

Comment:
Not the best form to do this, but this results in a much simpler fix that avoids a complete refactor of how the page manages the form and the update state of the source.